src/EventStore.Core/Services/Storage/ReaderIndex/IndexCommitter.cs

```diff
@@ -119,6 +119,16 @@ public class IndexCommitter : IIndexCommitter {
 							(result.RecordPostPosition - startPosition) * 100.0 / (buildToPosition - startPosition));
 						lastTime = DateTime.UtcNow;
 					}
+
+					if (processed % 1000000 == 0) {
+						if (_tableIndex.IsBackgroundTaskRunning) {
+							Log.Debug("Pausing ReadIndex Rebuild due to ongoing index merges.");
+							while (_tableIndex.IsBackgroundTaskRunning) {
```

Comment:
Would this result in large Event Store instances taking longer to come online because all merges must complete before the ReadIndex is available?

Reply:
@hayley-jean it could happen if all of the following conditions are true:
i) MaxMemTableSize was changed by the user to a value greater than 1M
ii) There are more than 1M events to rebuild when the node starts up
iii) When the node was shut down, there was a pending merge that was not completed.

We could ensure it happens only when the index is behind with something like:
`if(processed % Math.Max(MaxMemTableSize, 1000000) == 0`

Reply:
@hayley-jean i've fixed it in this commit: cfaf4cbc0cc344d95553d62bd0d06d0392702efb
The index rebuild will be paused during merges only if it's a full index rebuild.